azure/scope/clients.go

```diff
@@ -138,6 +138,8 @@ func (c *AzureClients) getSettingsFromEnvironment(environmentName string) (s aut
 	setValue(s, "AZURE_AD_RESOURCE")
 	if v := s.Values["AZURE_ENVIRONMENT"]; v == "" {
 		s.Environment = azureautorest.PublicCloud
+	} else if len(armEndpoint) > 0 {
+		s.Environment, err = azureautorest.EnvironmentFromURL(armEndpoint)
```

Comment:
Instead of adding an extra if check, is it possible to use `EnvironmentFromName`? It seems like `EnvironmentFromName` has some logic to handle AzureStackCloud, but not sure if it's what we want: https://github.com/Azure/go-autorest/blob/main/autorest/azure/environments.go#L300

Reply:
I have now refactored and done some initial testing to incorporate this suggestion. Now that we no longer need to pass the ARM endpoint around, this shrinks the size of this PR from changing 33 files to 12, and is more consistent with the existing patterns.

To use this pattern, users will need to run the manager with the `AZURE_ENVIRONMENT_FILEPATH` environment variable set and pointing to a json file containing their environment configuration. See https://github.com/Azure/go-autorest/blob/main/autorest/azure/environments.go#L300-L302 for more details.